plugins/jobtype/src/azkaban/jobtype/connectors/TeradataToHdfsJobRunnerMain.java

```diff
@@ -0,0 +1,105 @@
+import azkaban.utils.Props;
+import azkaban.jobExecutor.AbstractProcessJob;
+import azkaban.jobtype.*;
+import azkaban.jobtype.javautils.JobUtils;
+
+import com.teradata.hadoop.tool.TeradataImportTool;
+
+import static org.apache.hadoop.security.UserGroupInformation.HADOOP_TOKEN_FILE_LOCATION;
+
+public class TeradataToHdfsJobRunnerMain {
+  private static final int DEFAULT_NO_MAPPERS = 8;
+
+  private final Properties _jobProps;
+  private final TdchParameters _params;
+  private final Logger _logger;
+
+  public TeradataToHdfsJobRunnerMain() throws FileNotFoundException, IOException {
+    _logger = JobUtils.initJobLogger();
+    _jobProps = HadoopSecureWrapperUtils.loadAzkabanProps();
+    _jobProps.put(HadoopSecurityManager.ENABLE_PROXYING, "true"); //Always headless account
+
+    Props props = new Props(null, _jobProps);
+    HadoopConfigurationInjector.injectResources(props);
+    UserGroupInformation.setConfiguration(new Configuration());
+    props.getString(HadoopSecurityManager.USER_TO_PROXY); //Check required field.
+
+    _params = TdchParameters.builder()
+                            .mrParams(TdchConstants.MAP_REDUCE_PARAMS)
+                            .libJars(props.getString(TdchConstants.LIB_JARS_KEY))
+                            .tdJdbcClassName(TdchConstants.TERADATA_JDBCDRIVER_CLASSNAME)
+                            .teradataHostname(props.getString(TdchConstants.TD_HOSTNAME_KEY))
+                            .fileFormat(_jobProps.getProperty(TdchConstants.HDFS_FILE_FORMAT_KEY))
+                            .fieldSeparator(_jobProps.getProperty(TdchConstants.HDFS_FIELD_SEPARATOR_KEY))
+                            .jobType(TdchConstants.TDCH_JOB_TYPE)
+                            .userName(props.getString(TdchConstants.TD_USERID_KEY))
+                            .tdPassword(String.format(TdchConstants.TD_WALLET_FORMAT, props.getString(TdchConstants.TD_USERID_KEY)))
+                            .avroSchemaPath(_jobProps.getProperty(TdchConstants.AVRO_SCHEMA_PATH_KEY))
+                            .sourceTdTableName(_jobProps.getProperty(TdchConstants.SOURCE_TD_TABLE_NAME_KEY))
+                            .sourceQuery(_jobProps.getProperty(TdchConstants.SOURCE_TD_QUERY_NAME_KEY))
+                            .targetHdfsPath(props.getString(TdchConstants.TARGET_HDFS_PATH_KEY))
+                            .tdRetrieveMethod(_jobProps.getProperty(TdchConstants.TD_RETRIEVE_METHOD_KEY))
+                            .numMapper(DEFAULT_NO_MAPPERS)
+                            .build();
+  }
+
+  public void run() throws IOException, InterruptedException {
+    String jobName = System.getenv(AbstractProcessJob.JOB_NAME_ENV);
+    _logger.info("Running job " + jobName);
+
+    String tokenFile = System.getenv(HADOOP_TOKEN_FILE_LOCATION);
+    UserGroupInformation proxyUser =
+        HadoopSecureWrapperUtils.setupProxyUser(_jobProps, tokenFile, _logger);
+
+    proxyUser.doAs(new PrivilegedExceptionAction<Void>() {
```

Comment:
Does this code handle the scenario when proxying is not required?
Note how Spark has an if/else statement depending on whether shouldProxy() or not 

https://github.com/azkaban/azkaban-plugins/blob/master/plugins/jobtype/src/azkaban/jobtype/HadoopSecureSparkWrapper.java#L58